src/App.jsx

```diff
@@ -51,15 +52,34 @@ export function App() {
 
 			/** Finally, we update our React state. */
 			setData(nextData);
+			console.log(nextData);
 		});
 	}, [listToken]);
 
 	const handleCreateList = () => {
 		if (listToken) return;
-
 		setListToken(generateToken());
 	};
 
+	const handleJoinList = (token) => {
+		streamListItems(token, (snapshot) => {
+			const nextData = getItemData(snapshot);
+			if (nextData.length === 0) {
+				setErrorMsg('List does not exist!');
+			} else {
+				setData(nextData);
+				setListToken(token);
+			}
+		});
+	};
+	useEffect(() => {
```

Comment:
An interesting topic that is also a big debate since hooks appeared in react is when useEffect should be used. I personally tend to avoid it more often than not except if I need to. Since hooks first came into the foreground people tried to emulate the behaviour their class components had with the new hooks but react by itroducing them, also introduced a new mental model and many times people tend to useEffect when they could do things the more "Reactish" way. Don't change my word for granted though...I'd suggest just go take a look on the matter :) Here's also an article by react : https://react.dev/learn/you-might-not-need-an-effect

Reply:
This was really helpful 👏 Will definitely have it in mind going forward!